pkg/skaffold/jib/jib.go

```diff
@@ -88,6 +88,45 @@ type filesLists struct {
 // watchedFiles maps from project name to watched files
 var watchedFiles = map[string]filesLists{}
 
+// GetDependencies returns a list of files to watch for changes to rebuild
+func GetDependencies(ctx context.Context, workspace string, artifact *latest.JibArtifact) ([]string, error) {
+	t, err := DeterminePluginType(workspace, artifact)
+	if err != nil {
+		return nil, err
+	}
+
+	switch t {
+	case JibMaven:
+		return getDependenciesMaven(ctx, workspace, artifact)
+	case JibGradle:
+		return getDependenciesGradle(ctx, workspace, artifact)
+	default:
+		return nil, errors.Errorf("Unable to determine Jib builder type for %s", workspace)
+	}
+}
+
+// DeterminePluginType tries to determine the Jib plugin type for the given artifact.
+func DeterminePluginType(workspace string, artifact *latest.JibArtifact) (PluginType, error) {
+	// check if explicitly specified
+	if artifact != nil {
+		if t := PluginType(artifact.Type); t.IsKnown() {
+			return t, nil
+		}
+	}
+
+	// check for typical gradle files
+	for _, gradleFile := range []string{"build.gradle", "gradle.properties", "settings.gradle", "gradlew", "gradlew.bat", "gradlew.cmd"} {
```

Comment:
wonder if we should be searching for kotlin config files: `build.gradle.kts`?

Reply:
That's a good point.  Opened #2820 to address it separately as `skaffold init` support needs to be updated too.